src/lib.rs

```diff
@@ -173,7 +149,12 @@ mod test {
     fn all_signals() {
         let signals = [Signal::Hup, Signal::Int, Signal::Quit, Signal::Abrt, Signal::Term];
         let (tx, rx) = sync_channel(0);
-        super::set_handler(&signals, move |signals| tx.send(signals.to_owned()).unwrap());
+        let mut signal_count = 0;
+        super::set_handler(&signals, move |signals| {
+            signal_count += signals.len();
+            println!("Handled {} signals", signal_count);
```

Comment:
Hi! Is this "println!" here for a reason or it is just kind of debugging forgotten?

Reply:
It's there for a reason, but not a great one :) I wanted to prove the handler could be `FnMut` by testing with one that mutates captured variables, so I added the `signal_count` variable - then rust complained that it was unreferenced (since the code only increments it).

I added the printf to "use" the variable (and you can see the output with `cargo test -- --nocapture`) - I considered sending the current value out of the closure on a new channel and asserting the current count on the outside (as the test already does for the `signals`), but that's not really testing anything meaningful.

Reply:
Ah, okay, I got it :)

I doubt that the handler could really be `FnMut`, I have a strong feeling that we cannot directly mutate anything from another (detached) thread via `&mut` reference. 

For example consider the main thread has exited before the one in `set_handler`, then any access from it will trigger segmentation fault.

Reply:
`FnMut` allows mutation of captured variables, which is independent to accessing `&mut` references.  The use case I wanted was something like:

```
async fn main() {
  let (rx, tx) = tokio::sync::oneshot();
  let mut tx = Some(tx);
  set_handler(&signals, move |_| {
    if let Some(tx) = tx.take() {        // `take` needs `FnMut`
      tx.send("shutdown now please");
    }
  })

  tokio::spawn(...); // etc.

  rx.await;

  info!("Shutting down now");
}
```

As the handler is `'static` it can't capture references (mutable or otherwise) which prevents the segfault you were thinking about.

Reply:
okay that sounds perfectly reasonable!